lib/qless/worker/base.rb

```diff
@@ -50,11 +50,17 @@ def initialize(reserver, options = {})
       # USR1: Kill the forked children immediately, continue processing jobs.
       # USR2: Pause after this job
       # CONT: Start processing jobs again after a USR2
+      #  HUP: Print current stack to log and continue
       def register_signal_handlers
         # Otherwise, we want to take the appropriate action
         trap('TERM') { exit! }
         trap('INT')  { exit! }
         begin
+          trap('HUP') { log_stack_trace }
+        rescue ArgumentError
```

Comment:
On which platforms is `HUP` not supported?

For the other signals, below, I noticed that there is one rescue for all of them.  Rescuing it individually here is nice because the other signal handlers can still be registered if `HUP` is not supported.  It makes me think that maybe it's worth extracting a `try_to_trap` helper method that encapsulates the `trap; rescue` pattern so that each registration is handled individually so that all the ones that can be registered are.  Thoughts?